model/custom_status.go

```diff
@@ -15,9 +16,21 @@ const (
 	MaxRecentCustomStatuses  = 5
 )
 
+var ValidCustomStatusDuration = map[string]bool{
+	"dont_clear":     true,
```

Comment:
Should we handle the empty string as "don't clear" instead of making it explicit? As a user of the REST API, it would seem like duration should be optional.

Reply:
I think we should handle the default duration in the same way for both the API and slash commands. So we can either set default duration to "today" or "don't clear" in both API and slash commands? What do you think?

Reply:
My proposal would be to treat the empty string as "don't clear" -- as a user of the REST API, if I forget to specify this field, I'd love to have the behaviour of permanently setting the status. I can then opt-in to the auto-expiry by adding the field and providing it to the REST API.